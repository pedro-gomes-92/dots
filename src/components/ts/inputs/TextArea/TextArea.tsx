import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { BaseInputProps, BaseInput } from '../BaseInput';

export interface TextAreaProps extends BaseInputProps {
  hasFixedSize?: boolean;
  rows?: number;
}

export const TextArea: FunctionComponent<TextAreaProps> = ({
  className,
  hasFixedSize,
  rows,
  attribute,
  ...rest
}: TextAreaProps): JSX.Element => {
  let attributes = { ...attribute };
  if (rows) {
    attributes = { rows, ...attributes };
  }

  return (
    <BaseInput
      className={classnames('textarea', { 'has-fixed-size': hasFixedSize }, className)}
      attribute={attributes}
      {...rest}
    />
  );
};

TextArea.defaultProps = {
  tag: 'textarea',
  hasFixedSize: false,
};
