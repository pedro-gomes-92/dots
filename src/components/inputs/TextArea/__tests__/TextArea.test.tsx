import React from 'react';
import { render } from '@testing-library/react';

import { TextArea } from '../TextArea';

describe('Given <TextArea />', (): void => {
  let container: HTMLElement;
  describe('When initializes', (): void => {
    it('Then should match the snapshot', (): void => {
      container = render(<TextArea />).container;
      expect(container).toMatchSnapshot();
    });

    describe('And uses rows', (): void => {
      it('Then should match the snapshot', (): void => {
        container = render(<TextArea rows={10} />).container;
        expect(container).toMatchSnapshot();
      });
    });

    describe('And uses hasFixedSize', (): void => {
      beforeEach((): void => {
        container = render(<TextArea hasFixedSize />).container;
      });

      it('Then should match the snapshot', (): void => {
        expect(container).toMatchSnapshot();
      });

      it('And should not be resizable', (): void => {
        expect(container.querySelector('textarea').classList).toContainEqual('has-fixed-size');
      });
    });
  });
});
