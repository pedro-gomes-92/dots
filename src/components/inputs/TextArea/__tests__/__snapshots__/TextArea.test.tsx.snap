// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <TextArea /> When initializes And uses hasFixedSize Then should match the snapshot 1`] = `
<div>
  <textarea
    class="input textarea has-fixed-size"
  />
</div>
`;

exports[`Given <TextArea /> When initializes And uses rows Then should match the snapshot 1`] = `
<div>
  <textarea
    class="input textarea"
    rows="10"
  />
</div>
`;

exports[`Given <TextArea /> When initializes Then should match the snapshot 1`] = `
<div>
  <textarea
    class="input textarea"
  />
</div>
`;
